src/alert/actions-wrapper/index.tsx

```diff
@@ -0,0 +1,49 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import styles from './styles.css.js';
+import InternalButton, { InternalButtonProps } from '../../button/internal';
+
+function createActionButton(
+  testUtilClasses: ActionsWrapperProps['testUtilClasses'],
+  action: React.ReactNode,
+  buttonText: React.ReactNode,
+  onButtonClick: InternalButtonProps['onClick']
+) {
+  if (!action && buttonText) {
+    action = (
+      <InternalButton className={testUtilClasses.actionButton} onClick={onButtonClick} formAction="none">
+        {buttonText}
+      </InternalButton>
+    );
+  }
+  return action ? <div className={testUtilClasses.action}>{action}</div> : null;
+}
+
+interface ActionsWrapperProps {
+  action: React.ReactNode;
+  discoveredAction: React.ReactNode;
+  buttonText: React.ReactNode;
+  onButtonClick: InternalButtonProps['onClick'];
+  testUtilClasses: { action: string; actionButton: string };
+}
+
+export function ActionsWrapper({
+  testUtilClasses,
+  action,
+  discoveredAction,
+  buttonText,
+  onButtonClick,
+}: ActionsWrapperProps) {
+  const actionButton = createActionButton(testUtilClasses, action, buttonText, onButtonClick);
+  if (!actionButton && !discoveredAction) {
+    return null;
+  }
+
+  return (
+    <div className={styles.root}>
```

Comment:
This is not a space between component, because it creates additional wrapping nodes and breaks even spacing when either `actionButton` or `discoveredAction` is missing

Since this is a non-public component, we can compromise CSS scoping a bit and flatten the dom